Project/core/Reservation.cs

```diff
@@ -143,58 +182,25 @@ public double CostPeople()
 
             foreach (Room room in rooms)
             {
-                cost = cost + CostRoom(room);      
+                cost += CostRoom(room);      
             }
 
-            return ((cost/NumberOfPeople()) * numberOfAdults * Contig.priceForPerson + (cost / NumberOfPeople()) * numberOfChildren * Contig.priceForChild + (cost / NumberOfPeople()) * numberOfBabies * Contig.priceForBabies);
+            return ((cost/NumberOfPeople()) * adults * Config.priceForPerson + (cost / NumberOfPeople()) * children * Config.priceForChild + (cost / NumberOfPeople()) * babies * Config.priceForBabies);
         }
 
-        public double Cost()
+        public override double Amount()
         {
             double cost = CostPeople();
             
             if(rooms.Count() >= 3)
-                cost = cost * Contig.priceGroups;
+                cost = cost * Config.DISCOUNT_GROUPS;
             if (Days() >= 14)
-                cost = cost * Contig.priceLongStay;
-
-            return cost;
-        }
+                cost = cost * Config.DISCOUNT_LONG_STAY;
 
-        public int Advance()
-        {
-            return (int)(Cost() * Contig.priceAdvances);
-        }
+            if(isAdvance == true)
+                return cost - Advance();
```

Comment:
Dalej masz pętlę rekurencyjną Amount <-> Advance.
Jak spróbujesz to uruchomić z `isAdvance == true` to się program zawiesi.